pkg/splunk/enterprise/afwscheduler.go

```diff
@@ -781,12 +776,191 @@ podCopyPhase:
+func (pplnPhase *PipelinePhase) installWorkerHandler(ppln *AppInstallPipeline, handlerWaiter *sync.WaitGroup, installTracker []chan struct{}) {
+	scopedLog := log.WithName("installWorkerHandler")
+	defer handlerWaiter.Done()
+
+installHandler:
+	for {
+		select {
+		case installWorker, channelOpen := <-pplnPhase.msgChannel:
+			if !channelOpen {
+				// Channel is closed, so, do not handle any more workers
+				scopedLog.Info("worker channel closed")
+				break installHandler
+			}
+
+			if installWorker != nil {
+				podExecClient := splutil.GetPodExecClient(installWorker.client, installWorker.cr, installWorker.targetPodName)
+				podID, _ := getOrdinalValFromPodName(installWorker.targetPodName)
+
+				ctxt := getLocalScopePlaybookContext(installWorker, installTracker[podID], podExecClient)
+				if ctxt != nil {
+					installWorker.waiter.Add(1)
+					go ctxt.runPlaybook()
+				} else {
+					<-installTracker[podID]
+					scopedLog.Error(nil, "unable to get the local scoped context. app name %s", installWorker.appDeployInfo.AppName)
+				}
+			} else {
+				// This should never happen
+				scopedLog.Error(nil, "invalid worker reference")
+			}
+
+		default:
+			time.Sleep(1 * time.Second)
+		}
+
+		time.Sleep(200 * time.Millisecond)
+	}
+
+	for {
+		if needToRunClusterScopedPlaybook(ppln) {
+			targetPodName := getApplicablePodNameForAppFramework(ppln.cr, 0)
+			podExecClient := splutil.GetPodExecClient(ppln.client, ppln.cr, targetPodName)
+
+			// sgontla: can we just pass the CR???
+			ctxt := getClusterScopePlaybookContext(ppln.client, ppln.cr, ppln, targetPodName, ppln.cr.GetObjectKind().GroupVersionKind().Kind, podExecClient)
+			if ctxt != nil {
+				ctxt.runPlaybook()
+			} else {
+				scopedLog.Error(nil, "unable to get the cluster scoped playbook context, kind: %s, name: %s", ppln.cr.GroupVersionKind().Kind, ppln.cr.GetName())
+			}
+		} else {
+			break
+		}
+
+		// Sleep for a second before retry
+		time.Sleep(1 * time.Second)
+	}
+
+	// Wait for all the workers to finish
+	scopedLog.Info("Waiting for all the workers to finish")
+	pplnPhase.workerWaiter.Wait()
+	scopedLog.Info("All the workers finished")
+}
+
 // installPhaseManager creates install phase manager for the afw installation pipeline
 func (ppln *AppInstallPipeline) installPhaseManager() {
 	scopedLog := log.WithName("installPhaseManager")
 	scopedLog.Info("Starting Install phase manager")
+
+	var handlerWaiter sync.WaitGroup
+
 	pplnPhase := ppln.pplnPhases[enterpriseApi.PhaseInstall]
 
+	// Unlike other phases of the pipeline, Install phase is constrained by number of parallel installs per pod(ideally one)
+	// So, pick an install worker, if and only if there is no active install going on for a given pod.
+	// installWorkerPool is used to track the active installs for a given CR replica members
+	// Note:- In future, it will be much simplified to use a trylock(sync package mutex supports trylock in version 1.18)
+	replicas := *ppln.sts.Spec.Replicas
+	if replicas == 0 {
```

Comment:
As discussed, this was meant to cover MC's lack of replica config, but isn't actually needed as the replicas are filled internally - above code to be removed